### PR TITLE
Added Unix Implementations for omrsock_socket and omrsock_close in OMRSOCK API

### PR DESCRIPTION
**Pull request by caohaley**
- Added new error codes relating to socket functions
- Added Unix implementations to omrsock_socket and close
- Modified test cases to test omrsock_socket and close

Depends on PR: #4878 
Issue: #4795

### CODE REVIEW

File: port/unix/omrsock.c [hunk 1]
```diff
@@ -26,15 +26,151 @@
...
+map_sockettype_OMR_to_OS(int32_t OMRSockType)
+{
+	switch(OMRSockType)
+	{
+		case OMRSOCK_STREAM:
+			return OS_SOCK_STREAM;
+		case OMRSOCK_DGRAM:
+			return OS_SOCK_DGRAM;
+	}
+	return OS_SOCK_ANY;
```
Comment (rwy0717) [inline port/unix/omrsock.c]:
```suggestion
	switch (OMRSockType) {
	case OMRSOCK_STREAM:
		return OS_SOCK_STREAM;
	case OMRSOCK_DGRAM:
		return OS_SOCK_DGRAM;
	default:
		break;
	}
	return OS_SOCK_ANY;
```

File: port/unix/omrsock.c [hunk 2]
```diff
@@ -26,15 +26,151 @@
...
+/**
+ * @internal Map OMRSOCK API user interface socket type to the OS socket type, 
+ * which may be defined differently depending on operating system. 
+ *
+ * @param OMRSockType The OMR socket type to be converted.
+ *
+ * @return OS socket type on success, or OS_SOCK_ANY if none exists.
+ */
+static int32_t
+map_sockettype_OMR_to_OS(int32_t OMRSockType)
```
Comment (rwy0717) [inline port/unix/omrsock.c]:
```suggestion
get_os_socktype(int32_t omr_socktype)
```

File: fvtest/porttest/omrsockTest.cpp [hunk 3]
```diff
@@ -254,36 +255,39 @@ TEST(PortSockTest, getaddrinfo_and_freeaddrinfo)
...
 	ASSERT_EQ(rc, 0);
 
 	OMRPORTLIB->sock_getaddrinfo_length(OMRPORTLIB, &result, &length);
 	ASSERT_NE(length, 0);
 
-	OMRPORTLIB->sock_freeaddrinfo(OMRPORTLIB, &result);
+	for (uint32_t i = 0; i < length; i++) {
+		EXPECT_EQ(OMRPORTLIB->sock_getaddrinfo_family(OMRPORTLIB, &result, &family, i), 0);
+		EXPECT_EQ(OMRPORTLIB->sock_getaddrinfo_socktype(OMRPORTLIB, &result, &sockType, i), 0);
+		EXPECT_EQ(OMRPORTLIB->sock_getaddrinfo_protocol(OMRPORTLIB, &result, &protocol, i), 0);
```
Comment (rwy0717) [inline fvtest/porttest/omrsockTest.cpp]:
```suggestion
		EXPECT_EQ(OMRPORTLIB->sock_addrinfo_family(OMRPORTLIB, &result, &family, i), 0);
		EXPECT_EQ(OMRPORTLIB->sock_addrinfo_socktype(OMRPORTLIB, &result, &sockType, i), 0);
		EXPECT_EQ(OMRPORTLIB->sock_addrinfo_protocol(OMRPORTLIB, &result, &protocol, i), 0);
```

File: port/unix/omrsock.c (hunk 1 above)
Comment (caohaley) [reply port/unix/omrsock.c]: I added it to this PR with another small commit at the beginning.

File: fvtest/porttest/omrsockTest.cpp (hunk 3 above)
Comment (caohaley) [reply fvtest/porttest/omrsockTest.cpp]: If it's okay, can we leave this fix for now? The Fix Addrinfo PR has code that fixes part of this. I will update it and fix it completely once the addrinfo fix is merged.

**Comment by rwy0717** [comment]
@genie-omr build all

**Comment by rwy0717** [comment]
@genie-omr build all

**Comment by rwy0717** [comment]
@genie-omr build zos

**Comment by rwy0717** [comment]
@genie-omr build all

**Pull request merged**